sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -390,7 +390,8 @@ private[hive] class ClientWrapper(
       replace,
       holdDDLTime,
       inheritTableSpecs,
-      isSkewedStoreAsSubdir)
+      isSkewedStoreAsSubdir,
```

Comment:
It's great to see work done on this bug, but I don't think these changes to the `client` sub-package should be part of it. This code needs to work against different Hive versions, so you need something like #6627 here.

There are specific bugs to add support for different metastores (e.g. SPARK-8066, SPARK-8067). There isn't one for 1.2 specifically, so either we should file one, or you can choose to do it as part of this change - but you'll probably want to build on top of my PR for the client part...